base-template.sh

```diff
@@ -65,6 +65,9 @@ set_main_variables() {
     if [ "${CURRENT_GIT_DIR}" = "--git-common-dir" ]; then
         CURRENT_GIT_DIR=".git" # reset to a sensible default
     fi
+
+    # Git null ref
+    GIT_NULL_REF="0000000000000000000000000000000000000000"
```

Comment:
This is only used in `update_shared_hooks_if_appropriate` so I'd declare it there.